### PR TITLE
pad op

### PR DESCRIPTION
**Pull request by antinucleon**

### CODE REVIEW

File: src/operator/unary_op-inl.h [hunk 1]
```diff
@@ -0,0 +1,139 @@
+/*!
+ * Copyright (c) 2015 by Contributors
+ * \file unary_op-inl.h
+ * \brief
+ * \author Bing Xu
+*/
+#ifndef MXNET_OPERATOR_UNARY_OP_INL_H_
```
Comment (tqchen) [inline src/operator/unary_op-inl.h]: change name of this file, unary op was too broad. Maybe simply pad

File: src/operator/unary_op-inl.h [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  std::vector<mshadow::index_t> ret;
+  ret.push_back(shp[1]);
+  ret.push_back(shp[0]);
+  return TShape(ret.begin(), ret.end());
+}
+
+// -------------- operator
+
+struct PadParam : public dmlc::Parameter<PadParam> {
+  TShape pad_shape;
```
Comment (tqchen) [inline src/operator/unary_op-inl.h]: We should discuss what this parameter name should be. Ideally it should be consistent with common experience. Let us open an issue about this.

File: src/operator/unary_op-inl.h (hunk 1 above)
Comment (antinucleon) [inline src/operator/unary_op-inl.h]: We can put all kinds of simple unary op here...

Comment (tqchen) [inline src/operator/unary_op-inl.h]: This name was still too generic, consider use simple_unary_op for now then

**Pull request closed without merging**